src/clusterfuzz/_internal/system/archive.py

```diff
@@ -163,24 +303,26 @@ def has_extension(extensions):
   return ArchiveType.UNKNOWN
 
 
-def get_file_list(archive_path, archive_obj=None, file_match_callback=None):
+def get_file_list(archive_path, file_match_callback=None):
   """List all files in an archive."""
+  reader = get_archive_reader(archive_path)
   return [
-      f.name for f in iterator(
-          archive_path, archive_obj, file_match_callback, should_extract=False)
+      f.filename
+      for f in reader.list_files()
+      if not file_match_callback or file_match_callback(f.filename)
```

Comment:
This is the third instance of this pattern yet, how about extracting it to a method on `ArchiveReader`:

```py
class ArchiveReader(abc.ABC):

 # This is overridden by subclasses.
 @abc.abstractmethod
 def list_all_files(self) ...

def list_files(self, file_match_callback=None):
  if file_match_callback is None:
    return self.list_all_files()

  return [f for f in self.list_all_files() if file_match_callback(f.filename))]
```